NSS-Day-Cohort-69/tabloid-one-more-div
# Feature - Create Post

### Purpose
To allow Users to create a new Post

### Files Changed
- Added DateCreated property to `Comment.cs`, `CommenteDTO.cs`, `CommentForPostDTO.cs`, `PostDTO.cs`, `PostForListDTO.cs`, and `Post.cs`
- Added DateCreated Property, and updated logic for calculating FormattedPublicationDate in `PostForPostDetailsDTO.cs`
- Updated seed data for Posts and Comments in `TabloidDbContext.cs` to include the new DateCreated property
- Created initial Post DTO for creating a new Post in new `PostCreateDTO.cs` DTO class
- Added endpoint for creating a new Post, and updated previous endpoints to use new models logic in `PostController.cs`
- Added fetch for creating a new Post in `postManager.js`
- Created file, and wrote rendering and logic for New Post Form view in new `PostForm.jsx` component module
- Added routing for New Post Form view in `ApplicationViews.jsx`
- Added nav button for New Post Form view in `NavBar.jsx`

### To Test
Fetch, setup, and run according to project README, then confirm the following
- When logged in, there is a "New Post" button in the navbar
- Clicking on the "New Post" button navigates the User to the New Form Post view
- Attempting to submit a Post without a Title, Content, or a Category prevents the Post from being submitted
- Successfully submitting a Post navigates the User to the Posts List view
- Clicking "Cancel" in the New Post Form view navigates the User to the Posts List view

closes #41
